test/operatorsource/basic_test.go

```diff
@@ -1,146 +1,101 @@
 package operatorsource
 
 import (
-	"bytes"
 	"fmt"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
 	"os"
-	"os/exec"
-	"strings"
 	"testing"
-
-	"github.com/stretchr/testify/require"
+	"time"
 )
 
-const ShellToUse = "bash"
-
-func Shellout(command string) (string, string, error) {
-	var stdout bytes.Buffer
-	var stderr bytes.Buffer
-	cmd := exec.Command(ShellToUse, "-c", command)
-	cmd.Stdout = &stdout
-	cmd.Stderr = &stderr
-	err := cmd.Run()
-	return stdout.String(), stderr.String(), err
-}
-
-func Test_OperatorSource_oc_commands(t *testing.T) {
-
-	defer CleanUp(t)
+var (
+	Client               = NewTestClient()
+	namespace            = "openshift-operators"
+	subName              = "devconsole"
+	label                = "name=devconsole-operator"
+	subscription, suberr = Client.GetSubscription(subName, namespace)
+)
 
-	t.Run("login", func(t *testing.T) { Login(t) })
-	t.Run("subscription", func(t *testing.T) { Subscription(t) })
-	t.Run("install plan", func(t *testing.T) { InstallPlan(t) })
-	t.Run("operator pod", func(t *testing.T) { OperatorPod(t) })
-}
+func Test_OperatorSource(t *testing.T) {
+	pods, err := Client.GetPodByLabel(label, namespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer CleanUp(t, &pods.Items[0])
```

Comment:
I'm checking Pods length before returning it from GetPodByLabel(), so if pod isn't created it will return nil. you can check it here: https://github.com/redhat-developer/devconsole-operator/pull/213/files#diff-31828c90d3e18decf90b57b66bbe0b66R59

Reply:
@pratikjagrut What if there is the case where you get both `pod` and `error` values nil from `GetPodByLabel` ? wouldn't it fail in that case ? So just for safe side I think you should check array length.

Reply:
@akashshinde I have checked the Items length `if len(pods.Items) == 0`, but you are saying I should add an extra check `if pods == nil`. Okay that can be done. Thanks.